policies/coding-style.md

```diff
@@ -323,7 +323,17 @@ because it is a simple way to add valuable information for the reader.
 The name in the prototype declaration should match the name in the function
 definition.
 
-### Chapter 6.1: Extending existing functions
+### Chapter 6.1: Checking function aguments
+
+A _public_ function should verify that it's arguments are sensible.
+This includes, but is not limited to, verifying that:
+- non-optional pointer arguments are not NULL;
+- enumerated arguments have a meaningful value and
+- numeric arguments are within expected ranges.
+
+Where an argument is not sensible, an error should be returned.
```

Comment:
Is failure to comply with the coding style a bug? I wonder if something needs to be said about this in the stable release updates policy. I am in favour of this change but am wary about opening the flood gates for all the issues and PRs relating to missing NULL checks in stable releases (or even master).

Reply:
That's very good point and yeah, we should not treat failure to comply with coding style in an existing code a bug.

Reply:
BTW a PR can modify multiple policy documents at once so if we want, we can modify the stable release updates policy in this PR as well. Although maybe it could be mentioned in the coding style policy instead. I am not sure which is the better place.

Reply:
The discussions around this were clear that fixing existing APIs wasn't going to happen quickly.

We also don't consider other coding style deviations to be bugs as far as I'm aware.

I'll add a not a bug note in the stable releases policy.

Reply:
Saying code style changes are not a bug should be in the stable releases policy.